src/main/java/gov/nasa/pds/tools/validate/rule/pds4/TableDataContentValidationRule.java

```diff
@@ -340,6 +344,7 @@ public void validateTableDataContents() throws MalformedURLException,
         } else {
           LOG.debug("table instanceof TableCharacter: else");
           TableCharacter tc = (TableCharacter) table;
```

Comment:
@qchaupds per comments in review, we need to add additional logic in the code about for handling record delimiters for `Table_Delimited`:

- [ ] update code and add tests for table_delimited

right now we only handle this for `Table_Character` per your updates below.